docs/quick-release-notes/main.go

```diff
@@ -0,0 +1,314 @@
+}
+
+func splitAndEncodeURLPath(urlPath string) (string, string) {
+	urlPathSplit := strings.Split(urlPath, "/")
+	owner, name := url.QueryEscape(urlPathSplit[0]), url.QueryEscape(urlPathSplit[1])
+	return owner, name
+}
+
+func getReleases(authToken string, repoPath string) []listOfReleasesForUpdate {
+	repoOwner, repoName := splitAndEncodeURLPath(repoPath)
+	releasePath := fmt.Sprintf("https://api.github.com/repos/%v/%v/releases", repoOwner, repoName)
+	req, err := http.NewRequest("GET", releasePath, nil)
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "token "+authToken)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var eachRelease []listOfReleasesForUpdate
+	err = json.Unmarshal(body, &eachRelease)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return eachRelease
+}
+
+func getTagsFromAPI(tagsRepoPath string) []tagInfo {
+	// https://docs.github.com/en/rest/reference/repos#list-repository-tags
+	tagsRepoOwner, tagsRepoName := splitAndEncodeURLPath(tagsRepoPath)
+	pathToTags := fmt.Sprintf("https://api.github.com/repos/%v/%v/tags", tagsRepoOwner, tagsRepoName)
+	resp, err := http.Get(pathToTags)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	responseFromAPI := []byte(body)
+	var tags []tagInfo
+	err = json.Unmarshal(responseFromAPI, &tags)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return tags
+}
+
+func getChangeLogInfo(scanner *bufio.Scanner) changeLogInfo {
+	re, err := regexp.Compile(`## (?P<versionNumberAndDate>(?P<versionNumber>\d+\.\d+\.\d+).*)`)
+	if err != nil {
+		log.Fatal(err)
+	}
+	lineCount := 0
+	changeLogLength := 0
+	var versionLineNumbers []int
+	var changeLogText []string
+	var versionTitles []string
+	var versionNumbers []string
+	for scanner.Scan() {
+		changeLogLength++
+		changeLogText = append(changeLogText, scanner.Text())
+		if re.MatchString(scanner.Text()) {
+			hashVersionAndDate := re.FindStringSubmatch(scanner.Text())
+			idxVersionNumberAndDate := re.SubexpIndex("versionNumberAndDate")
+			idxVersionNumber := re.SubexpIndex("versionNumber")
+			versionNumberAndDate := hashVersionAndDate[idxVersionNumberAndDate]
+			versionNumber := hashVersionAndDate[idxVersionNumber]
+			versionTitles = append(versionTitles, versionNumberAndDate)
+			versionNumbers = append(versionNumbers, versionNumber)
+			versionLineNumbers = append(versionLineNumbers, lineCount)
+		}
+		lineCount++
+	}
+	if scanner.Err() != nil {
+		fmt.Println(scanner.Err())
+	}
+	return changeLogInfo{versionLineNumbers, changeLogText, changeLogLength, versionTitles, versionNumbers}
+}
+
+func createReleaseNotes(changeLogInfo changeLogInfo) []string {
+	re, err := regexp.Compile(`^\[.+\]: .+`)
```

Comment:
Nit: Rename `re` to something more descriptive like `linkReferenceDefinitionRE` (https://spec.commonmark.org/0.30/#link-reference-definitions).